Composer/plugins/runtimes/src/index.ts

```diff
@@ -0,0 +1,220 @@
+        // used to read bot project template from source (bundled in plugin)
+        await copyDir(sourcePath, localDisk, destPath, project.fileStorage);
+        const schemaDstPath = path.join(project.dir, 'schemas');
+        const schemaSrcPath = path.join(sourcePath, 'azurewebapp/schemas');
+        const customSchemaExists = fs.existsSync(schemaDstPath);
+        const pathsToExclude: Set<string> = new Set();
+        if (customSchemaExists) {
+          const sdkExcludePath = await localDisk.glob('sdk.schema', schemaSrcPath);
+          if (sdkExcludePath.length > 0) {
+            pathsToExclude.add(path.join(schemaSrcPath, sdkExcludePath[0]));
+          }
+        }
+        await copyDir(schemaSrcPath, localDisk, schemaDstPath, project.fileStorage, pathsToExclude);
+        const schemaFolderInRuntime = path.join(destPath, 'azurewebapp/schemas');
+        await removeDirAndFiles(schemaFolderInRuntime);
+        return destPath;
+      }
+      throw new Error(`Runtime already exists at ${destPath}`);
+    },
+    setSkillManifest: async (
+      dstRuntimePath: string,
+      dstStorage: IFileStorage,
+      srcManifestDir: string,
+      srcStorage: IFileStorage,
+      mode = 'azurewebapp' // set default as azurewebapp
+    ) => {
+      // update manifst into runtime wwwroot
+      if (mode === 'azurewebapp') {
+        const manifestDstDir = path.resolve(dstRuntimePath, 'azurewebapp', 'wwwroot', 'manifests');
+
+        if (await fs.pathExists(manifestDstDir)) {
+          await removeDirAndFiles(manifestDstDir);
+        }
+
+        if (await fs.pathExists(srcManifestDir)) {
+          copyDir(srcManifestDir, srcStorage, manifestDstDir, dstStorage);
+        }
+      }
+    },
+  });
+
+  composer.addRuntimeTemplate({
+    key: 'node-azurewebapp',
+    name: 'JS (preview)',
+    startCommand: 'node ./lib/webapp.js',
+    path: path.resolve(__dirname, '../../../../runtime/node'),
+    build: async (runtimePath: string, _project: any) => {
+      // do stuff
+      composer.log('BUILD THIS JS PROJECT');
+      const { installOut, installErr } = await exec('npm install --dev', {
+        cwd: runtimePath,
+        stdio: 'pipe',
+      });
+      const { install2Out, install2Err } = await exec('npm run build', {
+        cwd: runtimePath,
+        stdio: 'pipe',
+      });
+      composer.log('BUILD COMPLETE');
+    },
+    run: async (project: any, localDisk: IFileStorage) => {
+      // do stuff
+    },
+    buildDeploy: async (runtimePath: string, project: any, settings: any, profileName: string): Promise<string> => {
+      // do stuff
+      composer.log('BUILD THIS JS PROJECT');
+      const { installOut, installErr } = await exec('npm install', {
+        cwd: path.resolve(runtimePath, '../'),
+        stdio: 'pipe',
+      });
+      const { install2Out, install2Err } = await exec('npm run build', {
+        cwd: path.resolve(runtimePath, '../'),
+        stdio: 'pipe',
+      });
```

Comment:
Same with these. Also, refer to my previous comment about the return value of `exec`